receiver/signalfxreceiver/factory.go

```diff
@@ -76,5 +100,19 @@ func (f *Factory) CreateMetricsReceiver(
 ) (component.MetricsReceiver, error) {
 
 	rCfg := cfg.(*Config)
+
+	port, err := extractPortFromEndpoint(rCfg.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	// verify that the configured port is not 0
+	if port == 0 {
+		err = fmt.Errorf("endpoint with non-zero port must be enabled for %s receiver",
+			rCfg.Name(),
+		)
+		return nil, err
+	}
```

Comment:
Since this is actually validating the configuration, it probably can be done in a separate method. So, in this method we end up with something like

```
	if err := rCfg.validate(); err != nil {
		return nil, err
	}
```

and in `config.go` have

```
func (cfg *Config) validate() {
  	port, err := extractPortFromEndpoint(rCfg.Endpoint)
	if err != nil {
		return nil, err
	}

	// verify that the configured port is not 0
	if port == 0 {
		err = fmt.Errorf("endpoint with non-zero port must be enabled for %s receiver",
			rCfg.Name(),
		)
		return nil, err
	}
}
```

Reply:
yeah, I saw both in the code base -- this should be cleaned up. I only went with the current approach as the sapm receiver uses it as well. happy to go either way, let me know.

Reply:
I went ahead and changed both

Reply:
Didn't realize we have similar logic in SAPM receiver too. Thanks for updating.